src/netlink.rs

```diff
@@ -428,6 +434,24 @@ async fn ipv6(sock: &mut NlSocket, ifa_index: i32) -> Result<Option<Ipv6Addr>> {
         .map(Ipv6Addr::from))
 }
 
+async fn read_nameservers() -> Result<Vec<IpAddr>> {
+    let file = util::read_file("/etc/resolv.conf")
+        .await
+        .error("Failed to read /etc/resolv.conf")?;
+    let mut nameservers = Vec::new();
+
+    for line in file.lines() {
+        if let Some(ip) = line.strip_prefix("nameserver ") {
```

Comment:
This file doesn't have to be space separated just whitespace separated.

<details>
<summary>implementation spoilers</summary>

```diff
diff --git a/src/netlink.rs b/src/netlink.rs
index ca80892ff..f207a31fe 100644
--- a/src/netlink.rs
+++ b/src/netlink.rs
@@ -441,10 +441,10 @@ async fn read_nameservers() -> Result<Vec<IpAddr>> {
     let mut nameservers = Vec::new();
 
     for line in file.lines() {
-        if let Some(ip) = line.strip_prefix("nameserver ") {
-            let ip = ip.trim();
-            if let Ok(ip) = ip.parse() {
-                nameservers.push(ip);
+        let mut line_parts = line.split_whitespace();
+        if line_parts.next() == Some("nameserver") {
+            if let Some(ip) = line_parts.next() {
+                nameservers.push(ip.parse().error("Unable to parse ip")?);
             }
         }
     }
```
</details>